packages/xdl/src/Web.js

```diff
@@ -79,39 +121,158 @@ export async function onlySupportsWebAsync(projectRoot) {
   return false;
 }
 
-export async function ensureWebSupportAsync(projectRoot) {
+async function promptToAddWebPlatform() {
+  // clearConsole();
+  const question = {
+    type: 'confirm',
+    name: 'should',
+    message: `It appears you don't explicitly define "${chalk.underline(
+      `web`
+    )}" as one of the supported "${chalk.underline(`platforms`)}" in your project's ${chalk.white(
+      `app.json`
+    )}. \n  Would you like to add it?`,
+    default: true,
+  };
+  const answer = await inquirer.prompt(question);
+  return !!answer.should;
+}
+
+async function addWebPlatformToAppConfig(projectRoot) {
+  const { exp } = await readConfigJsonAsync(projectRoot);
+
+  let currentPlatforms = [];
+  if (Array.isArray(exp.platforms) && exp.platforms.length) {
+    currentPlatforms = exp.platforms;
+  }
+  if (currentPlatforms.includes('web')) {
+    return;
+  }
+
+  await ConfigUtils.writeConfigJsonAsync(projectRoot, { platforms: [...currentPlatforms, 'web'] });
+}
+
+function isUsingYarn(projectRoot) {
+  const yarnLockPath = path.resolve(projectRoot, 'yarn.lock');
```

Comment:
This false detected `expo/apps/native-component-list` as not using yarn. Can we try something like this?
```js
const workspaceRoot = findWorkspaceRoot(projectRoot);
if (workspaceRoot) {
  return fs.existsSync(path.join(workspaceRoot, 'yarn.lock'));
} else {
  return fs.existsSync(path.join(projectRoot, 'yarn.lock'));
}
```